lib/allegro/webapi/client.rb

```diff
@@ -12,15 +14,25 @@ class Client
       def initialize(options)
         @password = options.password
         @options = options
+        @timeout_counter = 0
       end
 
       def call(operation_name, locals = {})
-        raw_data = client.call(operation_name, locals)
+        Timeout.timeout(5) { raw_data = client.call(operation_name, locals) }
```

Comment:
5 seconds may not be enough. Also, please parameterise it either using an argument (preferably) or an environment variable.